hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -381,34 +376,40 @@ protected final List<Path> compact(final CompactionRequestImpl request,
       } else {
         Closeables.close(scanner, true);
       }
-      if (!finished && writer != null) {
-        abortWriter();
+      if (!finished) {
+        if (writer != null) {
+          abortWriter(writer);
+        }
+        // This signals that the target file is no longer written and can be cleaned up
+        completeCompaction(request);
```

Comment:
Clarifying: we only need to call completeCompaction in the exceptional case (compaction didn't finish normally)? And `commitWriter()` down below is doing the same kind of cleanup work that `completeCompaction()` here is doing?